akka-remote/src/main/resources/reference.conf

```diff
@@ -98,6 +98,11 @@ akka {
 
     # Acknowledgment timeout of management commands sent to the transport stack.
     command-ack-timeout = 30 s
+    
+    # The timeout for outbound associations to perform the handshake.
+    # If the transport is akka.remote.netty.tcp or akka.remote.netty.ssl
+    # the configured connection-timeout for the transport will be used instead.
```

Comment:
but isn't that same issue from the user's perspective? at least all sane users.

There is also a backwards compatibility aspect of this.
Let's say that someone has configured connection-timeout = 30s, then we can't suddenly introduce a new property that fail the connection attempt after 15s (which is the default for handshake-timeout)